VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/NullableToShorthand/NullableToShorthandAnalyzer.cs

```diff
@@ -64,17 +64,28 @@ private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
 
                 // First we look through the different nodes that can't be nested
                 // If nothing is found, we check if it's perhaps a standalone expression (such as an object creation without assigning it to an identifier)
-                var parentNode =
-                    context.Node.AncestorsAndSelf().FirstOrDefault(x => variableAncestorNodes.Contains(x.Kind())) ??
-                    context.Node.AncestorsAndSelf().OfType<ExpressionStatementSyntax>().FirstOrDefault();
+                SyntaxNode parentNode = null;
+                foreach (var node in context.Node.AncestorsAndSelf())
+                {
+                    if (variableAncestorNodes.Contains(node.Kind()))
+                    {
+                        parentNode = node;
+                    }
+                }
 
                 if (parentNode == null)
                 {
-                    return;
+                    parentNode = context.Node.AncestorsAndSelf().SyntaxNodeOfType<ExpressionStatementSyntax>(SyntaxKind.ExpressionStatement).FirstOrDefault();
+
+                    if (parentNode == null)
+                    {
+                        return;
+                    }
                 }
 
                 if (parentNode.Kind() == SyntaxKind.LocalDeclarationStatement)
                 {
+                    // ReSharper disable once PossibleInvalidCastException
```

Comment:
let's not add all these resharper annotations. I don't like library-specific things in the codebase unless we can't avoid it.